packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_masternode_create.test.ts

```diff
@@ -0,0 +1,230 @@
+
+  // ISSUE(canonbrother): nValue same as value, nTokenId same as tokenId, its inconsistent vout struct issue
+  // https://github.com/DeFiCh/ain/blob/c812f0283a52840996659121a755a9f723be2392/src/masternodes/mn_checks.cpp#L441-L442
+  txn.vout = txn.vout.map((v: any) => {
+    return {
+      nValue: v.value,
+      script: v.script,
+      nTokenId: v.tokenId
+    }
+  })
+  const encoded: string = OP_CODES.OP_DEFI_TX_CREATE_MASTER_NODE(createMasternode).asBuffer().toString('hex')
+  const expectedRedeemScript = `6a${encoded}`
+
+  const outs = await sendTransaction(container, txn)
+  expect(outs.length).toStrictEqual(2)
+  expect(outs[0].value).toStrictEqual(1)
+  expect(outs[0].n).toStrictEqual(0)
+  expect(outs[0].tokenId).toStrictEqual(0)
+  expect(outs[0].scriptPubKey.asm.startsWith('OP_RETURN 4466547843')).toBeTruthy()
+  expect(outs[0].scriptPubKey.hex).toStrictEqual(expectedRedeemScript)
+  expect(outs[0].scriptPubKey.type).toStrictEqual('nulldata')
+
+  expect(outs[1].value).toStrictEqual(2)
+  expect(outs[1].n).toStrictEqual(1)
+  expect(outs[1].tokenId).toStrictEqual(0)
+  expect(outs[1].scriptPubKey.type).toStrictEqual('witness_v0_keyhash')
+  expect(outs[1].scriptPubKey.reqSigs).toStrictEqual(1)
+  expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+  const masternodesAfter = await jsonRpc.masternode.listMasternodes()
+  const masternodesAfterLength = Object.keys(masternodesAfter).length
+  expect(masternodesAfterLength).toStrictEqual(masternodesBeforeLength + 1)
+})
+
+it('should be failed if address is P2SH, other than P2PKH AND P2WPKH', async () => {
+  const balance = await container.call('getbalance')
+  expect(balance >= 20000).toBeTruthy()
+
+  const address = await container.getNewAddress('', 'p2sh-segwit')
+  const addressDest: P2SH = P2SH.fromAddress(RegTest, address, P2SH)
+  const addressDestKeyHash = addressDest.hex
+
+  const createMasternode: CreateMasterNode = {
+    operatorType: 0x02,
+    operatorAuthAddress: addressDestKeyHash
+  }
+
+  const script = await providers.elliptic.script()
+
+  const txn: any = await builder.masternode.create(createMasternode, script)
+
+  // ISSUE(canonbrother): nValue same as value, nTokenId same as tokenId, its inconsistent vout struct issue
+  // https://github.com/DeFiCh/ain/blob/c812f0283a52840996659121a755a9f723be2392/src/masternodes/mn_checks.cpp#L441-L442
+  txn.vout = txn.vout.map((v: any) => {
+    return {
+      nValue: v.value,
+      script: v.script,
+      nTokenId: v.tokenId
+    }
+  })
+
+  const promise = sendTransaction(container, txn)
+  await expect(promise).rejects.toThrow(DeFiDRpcError)
+  await expect(promise).rejects.toThrow('CreateMasternodeTx: bad owner and|or operator address (should be P2PKH or P2WPKH only) or node with those addresses exists')
+})
+
+// ISSUE(canonbrother): createmasternode via sendrawtransaction will bypass balance checker??
+it.skip('should be failed as min 20k DFI is needed', async () => {
```

Comment:
@fuxingloh  @monstrobishi @surangap 
i realized i missed the balance checker test so i added it then i found this issue
i ady checked the ain, seems it doesn't have balance checker while createmasternode via sendrawtransaction
but can you guys help confirm again? :pray:

Reply:
test description says it all? `regtest` does not have `20K` limit

Reply:
ohh.. checked.. just need 2 for collateral